protocol-units/settlement/mcr/client/src/eth_client.rs

```diff
@@ -3,84 +3,61 @@ use crate::send_eth_tx::SendTxErrorRule;
 use crate::send_eth_tx::UnderPriced;
 use crate::send_eth_tx::VerifyRule;
 use crate::{CommitmentStream, McrSettlementClientOperations};
+use movement_types::BlockCommitment;
+use movement_types::{Commitment, Id};
+
+use alloy::pubsub::PubSubFrontend;
 use alloy_network::Ethereum;
+use alloy_network::EthereumSigner;
 use alloy_primitives::Address;
+use alloy_primitives::U256;
 use alloy_provider::fillers::ChainIdFiller;
 use alloy_provider::fillers::FillProvider;
 use alloy_provider::fillers::GasFiller;
 use alloy_provider::fillers::JoinFill;
 use alloy_provider::fillers::NonceFiller;
 use alloy_provider::fillers::SignerFiller;
-use std::array::TryFromSliceError;
-use std::str::FromStr;
-//use alloy_provider::fillers::TxFiller;
-use alloy_provider::{ProviderBuilder, RootProvider};
-use alloy_transport::Transport;
-use movement_types::{Commitment, Id};
-use std::marker::PhantomData;
-use tokio_stream::StreamExt;
-//use alloy_network::Network;
 use alloy_provider::Provider;
-use thiserror::Error;
-//use alloy_network::EthereumSigner;
-use alloy_primitives::U256;
-//use alloy_provider::ProviderBuilder;
-use alloy_sol_types::sol;
-//use alloy_transport_http::Http;
-use alloy::pubsub::PubSubFrontend;
-use alloy_network::EthereumSigner;
+use alloy_provider::{ProviderBuilder, RootProvider};
 use alloy_signer_wallet::LocalWallet;
+use alloy_sol_types::sol;
 use alloy_transport::BoxTransport;
 use alloy_transport_ws::WsConnect;
-use movement_types::BlockCommitment;
-use std::env;
 
-const MRC_CONTRACT_ADDRESS: &str = "0xBf7c7AE15E23B2E19C7a1e3c36e245A71500e181";
-const MAX_TX_SEND_RETRY: usize = 10;
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tokio_stream::StreamExt;
+
+use std::array::TryFromSliceError;
+
+const MCR_CONTRACT_ADDRESS: &str = "0xBf7c7AE15E23B2E19C7a1e3c36e245A71500e181";
+const MAX_TX_SEND_RETRIES: u32 = 10;
 const DEFAULT_TX_GAS_LIMIT: u128 = 10_000_000_000_000_000;
 
-#[derive(Clone, Debug)]
-pub struct McrEthSettlementConfig {
-	pub mrc_contract_address: String,
+/// Configuration of the MCR settlement client.
+///
+/// This structure is meant to be used in serialization.
+/// Validation is done by the builder interface of the [`Client`].
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
```

Comment:
This could be moved into a separate micro-crate to trim dependencies for suzuka-config. Keeping it here to ease reviewing for this PR.